api/src/convert/ring_mlsag.rs

```diff
@@ -3,33 +3,42 @@
 //! Convert to/from external::RingMLSAG
 
 use crate::{external, ConversionError};
-use mc_transaction_core::ring_signature::{CurveScalar, KeyImage, RingMLSAG};
+use mc_transaction_core::ring_signature::{CurveScalar, RingMLSAG};
 
-impl From<&RingMLSAG> for external::RingMLSAG {
+impl From<&RingMLSAG> for external::RingMlsag {
     fn from(source: &RingMLSAG) -> Self {
-        let mut ring_mlsag = external::RingMLSAG::new();
-        ring_mlsag.set_c_zero(external::CurveScalar::from(&source.c_zero));
         let responses: Vec<external::CurveScalar> = source
             .responses
             .iter()
             .map(external::CurveScalar::from)
             .collect();
-        ring_mlsag.set_responses(responses.into());
-        ring_mlsag.set_key_image(external::KeyImage::from(&source.key_image));
-        ring_mlsag
+        Self {
+            c_zero: external::CurveScalar::from(&source.c_zero).into(),
+            responses,
+            key_image: external::KeyImage::from(&source.key_image).into(),
+        }
     }
 }
 
-impl TryFrom<&external::RingMLSAG> for RingMLSAG {
+impl TryFrom<&external::RingMlsag> for RingMLSAG {
     type Error = ConversionError;
 
-    fn try_from(source: &external::RingMLSAG) -> Result<Self, Self::Error> {
-        let c_zero = CurveScalar::try_from(source.get_c_zero())?;
-        let mut responses: Vec<CurveScalar> = Vec::new();
-        for response in source.get_responses() {
-            responses.push(CurveScalar::try_from(response)?);
-        }
```

Comment:
There were a number of these blocks that created a vector and then pushed back into it, I chose to use an iterator and collect into a vec.